### ISSUE TITLE
[$250] [Waiting for payment - 2024-10-06] iOS - Login - White screen persists for some time after login

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#49880]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.41-1
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Launch New Expensify app
2. Log in with an account with many reports

## Expected Result:
There will be no white screen after login

## Actual Result:
White screen persists for some time after login
On production, there is white screen but it appears very briefly

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/127dec60-2e34-4b95-af30-0bae07ce10b2

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021840741226805286590</li>
        <li>Upwork Job ID: 1840741226805286590</li>
        <li>Last Price Increase: 2024-09-30</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

**Comment by roryabraham** [comment]
Going to venture a guess that this is a front-end blocker, not a back-end blocker

**Comment by Julesssss** [comment]
This doesn't occur every time on my account, but I can reproduce on `9.0.41.1` and `9.0.41.0`.Not `9.0.40.6`. however

There are a lot of issues in the checklist. I'm working through them but in the meantime have asked for [thoughts](https://expensify.slack.com/archives/C01GTK53T8Q/p1727694093521269) in Slack.

**Comment by truph01** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
White screen persists for some time after login


### What is the root cause of that problem?
- We return null when the data is loading:

https://github.com/Expensify/App/blob/86360745431edb355c91e5594c741a54e5f88581/src/libs/Navigation/AppNavigator/AuthScreens.tsx#L401-L402
### What changes do you think we should make in order to solve the problem?
- We should return ```<FullScreenLoadingIndicator />``` instead.

### What alternative solutions did you explore? (Optional)

**Comment by truph01** [comment]
### [Proposal updated](https://github.com/Expensify/App/issues/49880#issuecomment-2382875388)

**Comment by Julesssss** [comment]
Thanks @truph01. As this is a blocker I'm testing your proposal [here](https://github.com/Expensify/App/pull/49917).

**Comment by Julesssss** [comment]
> #49185

Yeah, in this case a simple fix might be preferable.

**Comment by Julesssss** [comment]
Yeah, I'm fixing it [here](https://github.com/Expensify/App/pull/49917). 

@truph01 we'll payout 50% for locating the regression.

**Comment by Julesssss** [comment]
Let's [revert](https://github.com/Expensify/App/pull/49922) to speed things up. We can settle on the fix when re-submitting the PR.

@allgandalf will review

**Comment by truph01** [comment]
> @truph01 we'll payout 50% 

Thanks @Julesssss

**Comment by truph01** [comment]
@luacmartins Should I have payment based on @Julesssss 's [comment](https://github.com/Expensify/App/issues/49880#issuecomment-2383016333)?

**Comment by Julesssss** [comment]
This was autoclosed. Reopening. @allgandalf also needs payment for the C+ review

**Comment by Nomanahmeda789** [comment]
The full screen loader have some issue.

**Comment by Nomanahmeda789** [comment]
Contributor details
Your Expensify account email: nomanhameda789@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/nomanahmed7

**Comment by blazejkustra** [comment]
The revert doesn't fully fix the bug, when the internet speed is really bad you can still reproduce the bug, for a brief second white screen is shown after logging in:

https://github.com/user-attachments/assets/c082d55a-26d9-4ade-973d-e1562d5262f7

**Comment by Julesssss** [comment]
> The revert doesn't fully fix the bug, when the internet speed is really bad you can still reproduce the bug

Thanks for sharing. It's far more noticeable, though, after these changes. With a medium-large sized the issue is not seen once reverting the changes.

**Comment by Julesssss** [comment]
Reverted, deploy blocker removed.

**Comment by blazejkustra** [comment]
@Julesssss Reverting it was the right call, but I think we still need to investigate why the white screen is displayed 🤔

**Comment by Julesssss** [comment]
Yes of course. I'll create a new issue for that, as it shouldn't be treated as a prior blocker.

**Comment by Julesssss** [comment]
Created [this issue](https://github.com/Expensify/App/issues/49934) to figure out the cause of the blank page issue.

**Comment by truph01** [comment]
@Julesssss Sorry, do I and @allgandalf have payment based on [comment](https://github.com/Expensify/App/issues/49880#issuecomment-2383151550)?

**Comment by Julesssss** [comment]
@allgandalf were you paid yet? I think probably not right?

@JmillsExpensify could you please pay out the following once this has been confirmed? Thanks

- @truph01 $125 for locating blocker (50%)
- @allgandalf $250 for review of revert PR

**Comment by johncschuster** [comment]
Looks like @allgandalf needs to be paid out via Upwork for the review. I'm taking care of that in [this issue](https://github.com/Expensify/App/issues/49926).

**Comment by Julesssss** [comment]
Oh right. 

@johncschuster would you mind taking care of the other payment there too?

> @truph01 $125 for locating blocker (50%)

**Comment by johncschuster** [comment]
Ugh, I missed ^that comment to issue payment for locating the blocker. I am so sorry, @truph01! Let me get an Upwork job going for that and I'll issue payment right away.

**Comment by johncschuster** [comment]
@truph01 can you accept this [offer on Upwork](https://www.upwork.com/nx/wm/offer/107227392) when you get a moment?

**Comment by johncschuster** [comment]
This has been paid! Thanks for flagging it, @truph01!
